src/V3Udp.cpp

```diff
@@ -0,0 +1,422 @@
+        AstNodeDType* const typep = nodep->findBitDType(m_inputNum, m_inputNum, VSigning::NOSIGN);
+        m_iFieldVarp = new AstVar{fl, VVarType::MODULETEMP, "tableline__ifield__udptmp", typep};
+        m_inputVars.back()->addNextHere(m_iFieldVarp);
+        AstVarRef* const iFieldRefp = new AstVarRef{fl, m_iFieldVarp, VAccess::WRITE};
+        auto itr = m_inputVars.begin();
+        // Relate the input vars with the input field var by concat
+        AstNodeExpr* contactp = new AstVarRef{fl, *itr, VAccess::READ};
+        while (++itr != m_inputVars.end()) {
+            contactp = new AstConcat{fl, new AstVarRef{fl, *itr, VAccess::READ}, contactp};
+        }
+        AstNodeStmt* const iFieldStmtp = new AstAssignW{fl, iFieldRefp, contactp};
+        AstNode* currentNode = iFieldStmtp;
+        // Process every table line.
+        iterateChildren(nodep);
+        // First for the sequent parts;
+        if (!m_alwaysIfMapp.empty()) {
+            for (auto itr = m_alwaysIfMapp.begin(); itr != m_alwaysIfMapp.end(); itr++) {
+                currentNode->addNextHere(itr->first);
+                currentNode = itr->first;
+            }
+            AstAssignW* const sequentAssignStmtp
+                = new AstAssignW{fl, new AstVarRef{fl, m_oFieldVarp, VAccess::WRITE},
+                                 new AstVarRef{fl, m_newOutVarp, VAccess::READ}};
+            currentNode->addNextHere(sequentAssignStmtp);
+            currentNode = sequentAssignStmtp;
+        }
+        // Then for the comb parts.
+        if (m_comboNodeStmtp) {
+            // Use the always block to realize the UDP table.
+            // For the mixtrigger, use the always latch.
+            VAlwaysKwd alwaysKwd;
+            if (!m_alwaysIfMapp.empty()) alwaysKwd = VAlwaysKwd::ALWAYS_LATCH;
+            currentNode->addNextHere(new AstAlways{fl, alwaysKwd, nullptr, m_comboNodeStmtp});
+        }
+        // For initial block, initialize the tmp out var.
+        if (m_outputInitVerfp && m_newOutVarp) {
+            m_outputInitVerfp->replaceWith(
+                new AstVarRef{nodep->fileline(), m_newOutVarp, VAccess::WRITE});
+            VL_DO_DANGLING(pushDeletep(m_outputInitVerfp), m_outputInitVerfp);
+        }
+
+        nodep->replaceWith(iFieldStmtp);
+        m_trigAlwaysMapp.clear();
+        m_alwaysIfMapp.clear();
+        m_newOutVarp = nullptr;
+        VL_DO_DANGLING(pushDeletep(nodep), nodep);
+    }
+    void visit(AstUdpTableLine* nodep) override { processLogic(nodep); }
+    void visit(AstNode* nodep) override { iterateChildren(nodep); }
+    // For logic processing.
+    bool isEdgeTrig(std::string& valName) {
+        auto isNum = [](char str) {
+            if (str == '1' || str == '0') return true;
+            return false;
+        };
+        if (valName == "r" || valName == "f" || valName == "R" || valName == "F") return true;
+        if (valName.length() == 4) {
+            std::string valNum = valName.substr(1, 2);
```

Comment:
Likewise add `const` where can elsewhere too

```suggestion
            const std::string valNum = valName.substr(1, 2);
```